snap/snapenv/snapenv.go

```diff
@@ -97,7 +97,7 @@ func userEnv(info *snap.Info, home string) map[string]string {
 	result := map[string]string{
 		"SNAP_USER_COMMON": info.UserCommonDataDir(home),
 		"SNAP_USER_DATA":   info.UserDataDir(home),
-		"XDG_RUNTIME_DIR":  info.UserXdgRuntimeDir(os.Geteuid()),
+		"XDG_RUNTIME_DIR":  info.UserXdgRuntimeDirGlobal(os.Geteuid()),
```

Comment:
Oh, I forgot to mention, this is considerably less risky because although we set XDG_RUNTIME_DIR in the environment, snapd did not create it. Because in practice the /run/user/uid dir is always there as part of session management applications don't create it, so snaps would have to opt into creating it. Therefore a simple test would be to use an existing snapd (ie, without this PR) and install and run snaps from the store and see if they create the /run/user/uid/snap.name.